### ISSUE TITLE
External ChaosDelegate in not active status

### ISSUE THREAD

**Issue by kumar-shantanu** [litmuschaos/litmus#3907]
**What happened**:
followed the instructions and installed the chaos centre and delegates, the external delegate is in not active state and looking at the log of the subscriber component, I see this, 

**time="2023-02-22T18:15:34Z" level=info msg="Connecting to wss://litmus.mylocal.domain/backend/query"
time="2023-02-22T18:15:35Z" level=fatal msg="Failed to established websocket connection" error="websocket: bad handshake"**

**What you expected to happen**:

I expected this external delegate to be in active status 

**Where can this issue be corrected? (optional)**

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

This is not a firewall issue, I can run curl from the same environment without any errors, 

**curl https://litmus.mylocal.domain/backend/query -k
{"errors":[{"message":"operation  not found"}],"data":null}/ $**

**Comment by Wojciech-Chilinski** [comment]
I had the same issue. In my case ingress was misconfigured. Defaults in litmus helm charts are wrong for the GKE ingresses. I was able to fix that by changing litmus ingress to pass `/query` directly to the backend, and editing `agent-config` configMap on agent side with `SERVER_ADDR: https://<your domain>/query`.

**Comment by kumar-shantanu** [comment]
Hi, @Wojciech-Chilinski  Thanks for your reply, are you using HTTPS and if yes are you using private CA for certificate provisioning? I changed the ingress the way you suggest but it's still not working. In my case, my ingress is configured to serve on HTTPS and I am using private CA here.

**Comment by jackdanieIs** [comment]
Yes, we are using SSL but with a publicly trusted cert. What is the error message now? Have you tried to check logs, what service receiving the `/query` calls? for sure it is related to the ingress misconfiguration: https://litmuschaos.github.io/litmus/experiments/troubleshooting/portal/#subscriber-is-crashing-with-the-error-dialwebsocket-bad-handshake

**Comment by Wojciech-Chilinski** [comment]
Yes, we are using SSL but with a publicly trusted cert. What is the error message now? Have you tried to check logs, what service receiving the `/query` calls? for sure it is related to the ingress misconfiguration: https://litmuschaos.github.io/litmus/experiments/troubleshooting/portal/#subscriber-is-crashing-with-the-error-dialwebsocket-bad-handshake

**Comment by kumar-shantanu** [comment]
Thanks @Wojciech-Chilinski Finally I got to the root of it, the ingress controller I am using needs to have this annotation to support websocket traffic, 

`nginx.org/websocket-services: "service-name"`

https://docs.nginx.com/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-annotations/

All good now finally. :)
